README.md

```diff
@@ -202,4 +206,89 @@ exit 1
 
 This example asset:
 - Responds to `_assetInfo` with the required JSON and type.
-- Accepts only a single argument (the JSON object) and outputs a valid JSON object when invoked.
\ No newline at end of file
+- Accepts only a single argument (the JSON object) and outputs a valid JSON object when invoked.
+
+---
+
+## Conjure Backcompat Asset
+
+The `backcompat` asset type enables backward compatibility checking of Conjure APIs.
+
+### Requirements
+
+**Type Declaration:**
+- The asset must respond to the `_assetInfo` probe with:
+  ```json
+  { "type": "backcompat" }
+  ```
+
+**Invocation:**
+- The asset is invoked with a JSON object that uses a discriminated union pattern based on operation type.
+
+**Check Backcompat Mode (`checkBackCompat`):**
+```json
+{
+  "type": "checkBackCompat",
+  "checkBackCompat": {
+    "currentIR": "/tmp/current-ir.json",
+    "project": "my-api",
+    "groupId": "com.example",
+    "projectConfig": {
+      "output-dir": "./pkg",
+      "server": true,
+      "ir-locator": { "type": "yaml", "locator": "./conjure" }
+    },
+    "godelProjectDir": "/path/to/project"
+  }
+}
+```
+
+**Accept Backcompat Breaks Mode (`acceptBackCompatBreaks`):**
+```json
+{
+  "type": "acceptBackCompatBreaks",
+  "acceptBackCompatBreaks": {
+    "currentIR": "/tmp/current-ir.json",
+    "project": "my-api",
+    "groupId": "com.example",
+    "projectConfig": {
+      "output-dir": "./pkg",
+      "server": true,
+      "ir-locator": { "type": "yaml", "locator": "./conjure" }
+    },
+    "godelProjectDir": "/path/to/project"
+  }
+}
+```
+
+### Usage
+
+**Running Backcompat Checks:**
+```bash
+# Check all projects
+./godelw conjure-backcompat
+
+# Check a specific project
+./godelw conjure-backcompat --project=my-api
+```
+
+**Accepting Backcompat Breaks:**
+```bash
+# Accept breaks for all projects
+./godelw conjure-accept-backcompat-breaks
+
+# Accept breaks for a specific project
+./godelw conjure-accept-backcompat-breaks --project=my-api
+```
+
+### Behavior
+
+- **Non-published projects**: Projects without a `group-id` configured or with `publish: false` are skipped.
+- **No backcompat asset**: If no backcompat asset is configured, the tasks are no-ops.
+- **Multiple assets**: Only one backcompat asset is supported. If multiple backcompat assets are configured, the plugin will fail with an error.
+
+### Error Handling
+
+- If the asset is invoked with more than one argument, it must immediately fail.
+- On success, the asset should output `{}` and exit with code 0.
+- On failure, the asset should output an error message to stdout and exit with a non-zero code.
```

Comment:
Behavior of "output empty JSON on success, but output non-JSON error message on failure" seems counter-intuitive as it is not symmetric.

Would expect either:
1. Output nothing on success with exit code 0, error message on failure with non-0 exit code (preferred)
2. Always output JSON, in which case success can either be empty or have a field like "success/fail" and failure would set a field like "errorMessage").